app/Http/Controllers/PilotPassportController.php

```diff
@@ -0,0 +1,193 @@
+            $achievements = PilotPassportAward::where('cid', $pilot->id)->get();
+            $privacy = intval($pilot->privacy);
+        }
+        return view('site.pilot_passport')->with('challenges', $challenges)->with('enrollments', $enrollments)
+            ->with('achievements', $achievements)->with('privacy', $privacy)->with('tab', $tab)
+            ->with('view_challenge', $pg);
+    }
+
+    public function enroll(Request $request) {
+        $valid_enrollment = true;
+        if (!is_numeric($request->challenge_id)) {
+            $valid_enrollment = false;
+        }
+        $p = auth()->guard('realops')->user();
+        if (!$p) {
+            return redirect('/pilot_passport/login');
+        }
+        $c = PilotPassport::find($request->challenge_id);
+        if (!$c) {
+            $valid_enrollment = false;
+        }
+        if (!$valid_enrollment) {
+            return redirect(route('pilotPassportIndex'))->withInput(['tab' => 'enrollments'])->with('error', 'Enrollment data invalid. Please contact wm@ztlartcc.org for assistance.');
+        }
+        $e = PilotPassportEnrollment::where('cid', $p->id)->where('challenge_id', $request->challenge_id)->get();
+        if ($e->isEmpty()) {
+            $e = new PilotPassportEnrollment;
+            $e->cid = $p->id;
+            $e->challenge_id = $c->id;
+            $e->save();
+            Mail::to($p->email)->send(new PilotPassportMail('enroll', $p, $c));
+            return redirect(route('pilotPassportIndex'))->withInput(['tab' => 'enrollments'])->with('success', 'You are now enrolled in the ZTL Pilot Passport program!');
+        }
+        return redirect(route('pilotPassportIndex'))->withInput(['tab' => 'enrollments'])->with('error', 'You are already enrolled in this challenge.');
+    }
+
+    public function setPrivacy(Request $request) {
+        $p = auth()->guard('realops')->user();
+        if (is_null($p)) {
+            return redirect(route('pilotPassportIndex'))->withInput(['tab' => 'settings'])->with('error', 'Unable to adjust privacy settings - Invalid CID.');
+        }
+        $p->privacy = $request->privacy;
+        $p->save();
+        return redirect(route('pilotPassportIndex'))->withInput(['tab' => 'settings'])->with('success', 'Your privacy preferences have been saved.');
+    }
+
+    public function purgeData(Request $request) {
+        if (strtolower($request->input('confirm_text')) != "confirm - purge all") {
+            return redirect()->back()->with('error', 'Data not purged. Please type in the required message to continue');
+        }
+        $p = auth()->guard('realops')->user();
+        $p->delete();
+        $enrollments = PilotPassportEnrollment::where('cid', $request->id);
+        foreach($enrollments as $e) {
+            $e->delete();
+        }
+        $logs = PilotPassportLog::where('cid', $request->id);
+        foreach($logs as $l) {
+            $l->delete();
+        }
```

Comment:
We shouldn't be using `$request->id` here. We should use `$pilot->cid`. Allowing `$request->id` would allow anyone to maliciously delete anyone's data by injecting some other id

Reply:
Good catch - updated